src/Fixer/Whitespace/NoExtraWhitespaceFixer.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * Remove extra whitespace buried inside of any SLOC.
+ *
+ * @author Tristan Strathearn <r3oath@gmail.com>
+ */
+final class NoExtraWhitespaceFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $index => $token) {
+            if ($token->isWhitespace(" \0\x0B")) {
+                $space = $token->getContent();
+
+                // Here we'll normalize all consecutive whitespace into a
+                // single whitespace character.
+                if (strlen($space) > 1) {
+                    $token->setContent(' ');
+                }
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Remove extra whitespace buried inside of any SLOC.';
+    }
+
+    public function getPriority()
+    {
+        // Make sure this is generally run before any other whitespace fixer.
```

Comment:
For now you are guessing the priority. Don't guess it. Grab the proper value and prove it's proper, like:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/master/tests/FixerFactoryTest.php#L193
First, the assertion will check if the relation between to fixer priorities are the one you want, then, integration test will prove that the fixers must run in given order to make their work properly. If you won't find any case for that you don't need to set the priority.